src/sources/lolalytics.js

```diff
@@ -0,0 +1,304 @@
+import Promise from 'bluebird';
+import cheerio from 'cheerio';
+import R from 'ramda';
+
+import ChampionifyErrors from '../errors.js';
+import { cl, request, shorthandSkills, trinksCon } from '../helpers';
+import Log from '../logger';
+import progressbar from '../progressbar';
+import store from '../store';
+import T from '../translate';
+
+const default_schema = require('../../data/default.json');  //  Riot JSON schema
+
+//  Item headings from lolalytics website
+const item_block_names = ['Starting Items', 'First Item', 'Boots', 'Second Item', 'Third Item', 'Fourth Item', 'Fifth Item'];
+
+export const source_info = {
+  name: 'Lolalytics',
+  id: 'lolalytics'
+};
+
+export function getVersion() {
+  return request('http://current.lolalytics.com/')
+    .then(body => {
+      const $c = cheerio.load(body);
+      const lolalytics_header = $c('body > div > h2').text();
+      const patchRegex = new RegExp('League of Legends Patch ([0-9].[0-9]*)');
+      const lolalytics_ver = patchRegex.exec(lolalytics_header)[1];
+      store.set('lolalytics_ver', lolalytics_ver);
+      return lolalytics_ver;
+    })
+    .catch(err => {
+      throw new ChampionifyErrors.RequestError('Can\'t get Lolalytics Version').causedBy(err);
+    });
+}
+
+function createChampionifyJson(riot_json, position, champion_name, split_sort) {
+  if (split_sort) {
+    position += `_${split_sort}`;
+  }
+
+  return {
+    champ: champion_name,
+    file_prefix: position,
+    riot_json: riot_json,
+    source: 'lolalytics'
+  };
+}
+
+function createRiotJson(parsed_data, position, champion_name, split_sort) {
+  const blocks = [];
+  const item_names = store.get('item_names');
+  const sortByWinRate = R.sortBy(R.prop('win'));
+  const sortByPickRate = R.sortBy(R.prop('pick'));
+  const build_sections = parsed_data.build_sections;
+  const skills = parsed_data.skills;
+
+  const build_sections_to_convert = [];
+
+  R.forEach(function(x) { if (R.contains(x.title, item_block_names)) build_sections_to_convert.push(x); }, build_sections); //  Filter out the rows we don't care about (champion counters, etc.)
+
+  function convertBuildSectionJsonToBlock(build_section_json, title) {
+    const items = [];
+
+    for (let k = 0; k < build_section_json.length; k++) {
+      let id = parseInt(item_names[build_section_json[k].name], 10);
+
+      id = (isNaN(id) ? 2003 : id); //  For biscuit
+
+      items.push({
+        count: 1,
+        id: id.toString()
+      });
+    }
+
+    const filtered_items = R.uniqBy(R.prop('id'), items); //  filter out duplicates, mainly just the second health pot from converting buscuits
```

Comment:
The issue with going along this route is that it doesn't show the `2x` and such on top of the item in game because you have count set to `1` rather then how many pots someone should actually be purchasing. You can actually get rid of your entire loop for counting items if you were to use something like [this](https://github.com/dustinblackman/Championify/blob/master/src/sources/koreanbuilds.js#L29-L40) instead. That'll take an array of item codes, and return objects that include `id` and the proper `count`. You're welcome to copy/paste it, I'll probably move it in to a helper at a later time.